### DISCUSSION TITLE
Вылет при загрузки уровня (новой игры). Когда была просто изменено имя в ogf

### DISCUSSION THREAD

**Discussion by energomuzik** [PavelBlend/blender-xray#431] [Q&A]
![дисскусия1](https://user-images.githubusercontent.com/92163825/136565871-95f98f4e-c58b-4ea9-926d-943b7c38aad0.jpg)
 Я просто импортировал файл ogf , ничего не делая экспортировал в ДРУГОЕ ИМЯ. 
![дискусия2](https://user-images.githubusercontent.com/92163825/136566182-a4e47ecc-8dca-4c70-bf53-01169f5761d3.png)
вписал имя в дринки, стрин тэйбл, энциклопедию эквипмента. Но при запуске игра вылетает без лога.  Меняю на прежнюю vodka игра нормально запускается , есть новая иконка водки, имя, описание. 
может в том , что там еще есть несколько файлов по ''обьему''
![дискусия3](https://user-images.githubusercontent.com/92163825/136566866-04914d2a-b346-4a27-82e2-72efbc0d60b0.png)
Второй вопрос. Вот допустим с именем мы проблему решим, по вашей подсказке. Я то хотел просто сменить текстуру у водки , сохранил с помощью паинтнет в dds , поменял вот тут путь и тут , экспортировал в ogf с другим именем, и когда загружаю и вот в игре ничего не загрузилось облом. =)   
![дискус4](https://user-images.githubusercontent.com/92163825/136568081-dea6ff08-2f2f-4ec9-8649-d7ac6603854d.png)
И главное то , что когда я снова загружаю эту ogf модель она нормально загружается!
![диск5](https://user-images.githubusercontent.com/92163825/136568370-0eb481fb-5afc-4511-9d7d-8b3bcaf9d76c.png)
Помогите если не трудно. =)

**Comment by PavelBlend** [comment]
@energomuzik какая версия игры? ЗП/ТЧ/ЧН? Или может мод?

**Reply by energomuzik** [reply]
сталкер тень чернобыля модификация prosectors project 1.6 ! эта моделька использовалась в анимации . они надобавляли  много анимаций.

**Comment by PavelBlend** [comment]
@energomuzik попробуй не делать реэкспорт ogf, а измени имя существующего файла. Если не будет вылета, то сделай реэкспорт ogf файла блендером, ничего не меняя при этом. Если тоже не вылетит, то потом уже пробуй менять конфиги. Возможно в конфиге опечатка. Движок может без лога из-за этого вылетать. В общем старайся за один раз менять что-то одно и запускай игру. Так легче будет понять, из-за чего вылет.

И какая итоговая цель? Заменить существующую модель бутылки или добавить новую? Я просто не могу понять, зачем править конфиги, string_table, энциклопедию и т.д.

**Comment by PavelBlend** [comment]
@energomuzik а в этом моде движок правленый? Если нет, то можешь скачать 1.0007 патч отсюда:
https://github.com/PavelBlend/xp-dev_xray/releases/tag/build-5578-demo-record

Он без лога не вылетает. И лог более подробный чем в 1.0006.

**Reply by energomuzik** [reply]
Блин, правленный вот вроде тут исходники. https://gitlab.com/Karlan/Prosectors
А для чего тогда делался мод. Там написано, что переработали скрипты , конфиги, для удобной работы. И да я даже начал кое чего понимать , в логике игры . Почему все так трудно. Ведь как бы вы уже подчти сделали простой способ добавить модель в игру. А ведь до этого надо было конвертировать модель  ogf в object , затем с помощью сдк сохранять . Кстати если  prosectors game data  скопировать в game data sdk , то и sdk прекратит работать, и игра внутри sdk тоже, когда оригинальная игра в sdk запускается нормально, и уровень редактируется, изменяется, конвертируется и записывается нормально. Значит и правда движок изменили.  Только вот как добавлять в игру модельки по старому или вы сможете доработать аддон? или он будет работать только с оригинальной игрой в простом нативном изменнии импорт экспорт ogf , а не мучится с конвертациями в object ?

**Reply by PavelBlend** [reply]
@energomuzik попробуй импортировать ogf модель в блендер, но не изменять её, а сразу сделать экспорт в object. Потом открой object в сдк и сделай экспорт в ogf. И замени полученный ogf на оригинальный. Вылета не должно быть. Я не думаю, что они изменили формат ogf моделей. Если вылета не будет, то можно будет думать дальше, в чём причина.

Вылет может быть из-за boneparts. Правда этот вылет с логом должен быть. Проверь, есть ли у скелета Bone Groups в блендере?

**Reply by energomuzik** [reply]
Взял импортировал оригинальную модель vodka.ogf в блендер, после этого сразу ничего не меняя жмакнул на экспортирование в x-ray object .  Открыл sdk для сталкера теня ченобыля sdk 0.4  левел эдитор,  пытаюсь заспавнить object vodka  он не добавляет и пишет ошибка! При том , что вот рядышком оригинальный object объект который был в сдк нормально загрузился.
![sdkй](https://user-images.githubusercontent.com/92163825/136644981-95aa4412-063b-49c2-ac35-a1591da24b4b.png)
Может это sdk не тот и надо sdk от зова припяти 0.7 версии? 
![xray2](https://user-images.githubusercontent.com/92163825/136645009-35a2fb7f-4c34-4a30-853d-00a145309f54.png)
 по поводу bonegroups  их нету! вроде! 
![lbcrec1](https://user-images.githubusercontent.com/92163825/136645061-2228cd4d-30f2-460b-b1f3-7ae4fe0e6aff.png)
может это версия вашего плагина неправильно установилась. До этого была ваша версия 1.5 мегабайт, я эту папку вырезал и сохранил в резерве, и заместо его положил новую версию она всего весит 0.8 мегабайта!  Запустил блендер и он нормально запустился. Для надежности убрал галочку с x-ray . Перезагрузил. Снова нажал на галочку x-ray. Выбрал все шеидеры . Может я не то выбрал. 
![блендеир](https://user-images.githubusercontent.com/92163825/136645208-d584d0cd-305d-4119-8530-282f3e4fe547.png)
Я еще раньше заметил , что ваш аддон требует  лтикс с расположением всех шеидеров в сдк, и именно сдк . А если человек хочет редактировать модельки в каком нибудь моде.  Файлы в prosectorse 1.6 были запакованы в db файлах. я их распаковал вот этим
![датаунпакер](https://user-images.githubusercontent.com/92163825/136645345-ced0f06e-f52d-4ee7-bf17-a58b35a715aa.png)

Может унпакер не тот?

**Comment by PavelBlend** [comment]
@energomuzik переименуй object файл с `vodka.ogf.object` на `vodka.object`. СДК требует, чтобы имя файла имело только одну точку.

> по поводу bonegroups их нету! вроде!

а эта модель имеет анимации? Если анимаций нет, то bone parts тоже не должно быть. А если есть, то bone part хранятся внутри ogf только в том случае, если он содержит анимации. Если анимаций в ogf нет, а они используются в omf файле, то bone part нужно импортировать из omf.

> До этого была ваша версия 1.5 мегабайт
там кеш добавляется. Поэтому размер папки выше.

**Reply by energomuzik** [reply]
С файлом object и манипуляции с ним прошли НОРМАЛЬНО. экспортировал файл из сдк в vodka.ogf и игра нормально запустилась. Только вот Я внимательнее почитал, и тут у файла водка сразу в его drink.ltx используется ТРИ анимации ogf vodka.ogf просто бутылка без бон партс и анимаций напротив visual, вторая ogf на против  hud  anm_vodka_hud ogf который я открыл и там уже есть руки бутылка водки и анимация, только не до конца проигрывается!!!!!!  ,  и третья анимация напротив item visual  динамика weapon wodkakazakihud.ogf где просто бутылка водки но УЖЕ с костями и разделена НА ДВЕ части крышечка и сама бутылка!. у которой есть анимация.  и она проигрывается ПОЛНОСТЬЮ, но она же без РУК! КАК ЭТО РАБОТАЕТ?!!! Что это за ФАЙЛОВАЯ БЮРОКРАТИЯ!!! один файл тут, отсылают в другое место, а тут анимация, ...
Как мне быть , как добавить новый ИТЕМ со старой анимацией ? Я например хочу добавить новый арт, у которой тоже есть анимация использования артефакта - его варка и активация. Как понимать вот эти три ogf файла и где корень проблемы добавления нового предмета с анимацией его поедания или активации арта. Там был еще файл omf eat_hand_animations , но при добавлении его к остальным ogf файлам сразу писалась ошибка в блендере.
![2](https://user-images.githubusercontent.com/92163825/136665669-4727296c-d119-4757-a330-9993a281cfbe.png)
![3](https://user-images.githubusercontent.com/92163825/136665674-58a29492-36a2-449a-969a-78366993a9a3.png)

**Comment by PavelBlend** [comment]
@energomuzik модель vodka.ogf - это модель без анимаций, которую ты видишь на земле, если выкинуть водку из рюкзака под ноги. А модель vodka_hud.ogf - это модель водки с руками с анимациями, которую ты видишь от первого лица. Я не знаю, как в этом моде устроена система hud-моделей. Возможно в файле vodka_hud.ogf нет рук, а есть только бутылка (так в ЗП сделано). В ТЧ без модов оружие (можно сделать и водку/артефакты) совмещены с моделью рук и находятся в одном ogf файле.

**Reply by energomuzik** [reply]
Тут накрутили немного по другому как вы рассказываете
![диск1](https://user-images.githubusercontent.com/92163825/136691346-09f8791e-c9ca-4a8f-988b-0274140eb27a.png)
![диск2](https://user-images.githubusercontent.com/92163825/136691354-6eac7252-7cab-46c3-a90a-1af35e21b52b.png)

![lbcr4](https://user-images.githubusercontent.com/92163825/136691342-e8e35886-009e-4c3e-8d87-84ddc779743a.png)
Что за ошибка  в логе? почему нельзя изменить визуал? как мне правильно собрать модель , что бы в блендере можно
было изменить анимацию. например добавить к старой , разворот бутылки с рукой этикеткой к себе, что бы посмотреть что пьешь =) Потому как тут в файле vodka_hud.ogf ФЭЙКОВАЯ бутылка с низким разрешением! для кого она сделана?

**Comment by PavelBlend** [comment]
@energomuzik модель не в той папке лежит. У тебя на скрине папка `weapon`, а в логе `weapons`. В конце `s` не хватает.

**Reply by energomuzik** [reply]
Спасибо не внимательность второе я =). заработало. Только вот теперь,  не могу понять где эти анимации менять. Ведь файл  vodka_hud нигде не указан в файле описании drink.ltx там. Да  есть  фэйейковая не полная  анимация в vodka_hud  Для кого она? может с помощью него делали анимации неполные и частями сохраняли в формате anm. Которые в файле дринков и указаны. Однако я нашел только положение камеры  по описанию drinks.ltx Где менять анимацию? анимация в редакторе совсем не такая как в игре.
![диск6](https://user-images.githubusercontent.com/92163825/136697407-c8999578-dfbf-426c-b55f-697be23288b9.png)

**Reply by PavelBlend** [reply]
anm анимации - это анимации камеры, когда гг пьяный и т. д.

Я не знаю, как устроены анимации в этом моде. Как в ЗП или как в ТЧ? Тебе просто нужно изучить готовые модели и конфиги и сделать по аналогии. Желательно начать с чего-то простого, если до этого ничего не создавал. Если опыта мало, то лучше делать изменения для ТЧ или ЗП. Так как для них уроков и статей больше. И они более изученные. На форумах легче найти ответ. А в этом моде возможно авторы изменили систему hud моделей.

Если в этом моде система hud как в зп, то поищи статьи по созданию оружия для зп.

**Reply by energomuzik** [reply]
Спасибо за ответ,  попробую! Задаваться не буду  , буду медленно но копать в правильном направлении. И вам удачи сделать хорошее дополнение к блендеру  !
